libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/state/ManeuverState.kt

```diff
@@ -0,0 +1,29 @@
+package com.mapbox.navigation.dropin.state
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+import com.mapbox.navigation.ui.maneuver.model.Maneuver
+import com.mapbox.navigation.ui.maneuver.model.ManeuverError
+import com.mapbox.navigation.ui.maneuver.model.ManeuverErrorFactory
+
+data class ManeuverState(
+    val isVisible: Boolean,
+    val navigationState: NavigationState,
+    val maneuver: Expected<ManeuverError, List<Maneuver>>
+) {
+    @ExperimentalMapboxNavigationAPI
+    companion object {
+        fun initial(): ManeuverState =
+            ManeuverState(
+                isVisible = false,
+                navigationState = NavigationState.Empty,
+                maneuver = ExpectedFactory.createError(
+                    ManeuverErrorFactory.buildManeuverError(
```

Comment:
I had to create a new `ManeuverErrorFactory` that would allow to construct this object. Since we have marked `*Factory` with @ExperimentalMapboxNavigationAPI, this has taken affect upstream to `MainActivity`. We need to make a decision if it's ok to add this tag to an Activity, or do we want to remove this tag from any `*Factory` objects that we create.
/cc @LukasPaczos

Reply:
@abhishek1508 we don't have to mark all classes that use the new experimental factory with `@ExperimentalMapboxNavigationAPI` (which then goes up and up the usage tree), we just need to tag the specific internal usages with `@OptIn(ExperimentalMapboxNavigationAPI::class)` which does not bubble up.